syntaxes/templ.tmLanguage.json

```diff
@@ -389,41 +386,53 @@
         }
       ]
     },
-    "empty-import-expression":{
-      "match": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*$",
-      "captures": {
-        "0": {
+    "import-expression": {
+      "patterns": [
+        {
+          "name": "start.import-expression.templ",
+          "begin": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(",
+          "beginCaptures": {
+            "0": {
+              "patterns": [
+                {
+                  "include": "#import-expression-start"
+                }
+              ]
+            }
+          },
+          "end": "(\\))",
+          "endCaptures": {
+            "1": {
+              "name": "punctuation.definition.end.bracket.round.go"
+            }
+          },
           "patterns": [
             {
-              "include": "#import-expression-start"
+              "include": "source.go"
             }
           ]
-        }
-      }
-    },
-    "import-expression": {
-      "begin": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*{$",
-      "beginCaptures": {
-        "0": {
+        },
+        {
+          "name": "children.import-expression.templ",
```

Comment:
This second pattern captures from the previous pattern's closing parenthesis and opening curly brace to the closing curly brace and marks syntax inside them with the `template-node`.

Reply:
Probably best to wait on merging this. Checked again real quick on the token inspector and found that this second pattern matching is matching against all `) {` sequences and not just ones that follow the opening `@functionName()` rule above.